### ISSUE TITLE
Latex vignettes

### ISSUE THREAD

**Issue by eddelbuettel** [r-universe-org/roadmap#40]
R supports more than one vignette format, but the R-Universe presentation more or less suggests that there is only one: html.  It would be somewhat more representative of the universe of CRAN packages at CRAN if all vignettes were included here too.

**Comment by eddelbuettel** [comment]
By doing what you are doing you are creating a second reality that does not exist. You make the set of documented packages appear (way) smaller than it is.  There is no indication that it is missing.  I don't think that is a positive step.

Could you maybe consider using an iframe or alike (you know webstacks better than I do...) to do something akin to this where I embed existing pdf in a markdown-made (with a different toolchain) page:  https://eddelbuettel.github.io/anytime/vignette/

**Comment by jeroen** [comment]
I tried adding tinytex on the build image, but in your case that did not help because you require the `includepdf` latex pkg.

By the way, given that in `anytime` and `Rcpp` you already include the pdf files in the package, you can easily make it work using a vignette engine that simply copies those `pdf` file, rather than creating a new pdf file that embeds the other one.

Henrik's R.rsp engine does exactly his, I use this in jsonlite as well to embed a paper, see: https://github.com/jeroen/jsonlite/blob/master/vignettes/json-opencpu.pdf.asis

**Comment by eddelbuettel** [comment]
Thanks for trying it.

But recall that this a discussion that has been had a dozen times: I prefer and recommend Mark's (Sweave wrapper) rather than Henrik's solution for a reason (namely that that my packages remain simpler smaller).   Asking me to change working vignettes that are accepted at CRAN is a little too strong in my book.  

Sweave vignettes are a standard feature of R and CRAN packages.  Deciding you will not support them is an odd choice given the "universe" aim here. 

Can you not just iframe around them at the presentation end and take them as given if your setup decides it does not want to rebuild them?  Otherwise maybe you need to farm out to a dedicated pdf vignette building facility.

**Comment by jeroen** [comment]
I don't see how your packages remain simpler and smaller, by taking the full latex system as a dependency, only to copy a pdf file. That is really a double standard.

For now we don't support heavy bloated deps including latex and Java. Maybe some time in the future.

**Comment by eddelbuettel** [comment]
Well pdf haters gonna hate, over here we like them -- as does R itself as I remind you again that you are shutting the default vignette mechanism of R packages out of your universe by choice.  

And I don't think that when aiming to be "universal" you should be taking sides and making preferences (and altering html vignettes by swapping / removing css files is another kettle of fish but I leave that for another time).

**Comment by jeroen** [comment]
Giving tinytex another try: https://github.com/r-universe-org/base-image/commit/57ecf0e99ab3287e1d53df9ae461e2ac3e6725c9

According to twitter, tinytex should be able to automatically install missing latex packages but maybe that only works for rmarkdown inputs: https://twitter.com/opencpu/status/1377221618242441219

**Comment by eddelbuettel** [comment]
I really appreciate you trying in order to fully represent CRAN packages.  

Some historical context: On the platforms I use (and which are likely close that your containers are based on) I have found `tinytex` to be not exactly helping (and even doing quite the opposite; even KH and I were once chasing a bug in a latex package it actively masked or camouflage by insisting on being in the middle). I tried to provide helpful feedback on that to its author but that went nowhere -- though all that was a while back.  I sometimes wished we could deploy a toggle to just switch `tinytex` off and fall back to a known, working, complete texlive setups but that suggestion too went nowhere.  

"In theory" your setup would be a perfect testbed for `tinytex`.  (Part of me still thinks it is actually inadvisable to waste resources on trying to rebuild infrastructure for something we _by existence of a CRAN package_ know builds there so why not just copy it but that too is your design choice.)  So in an ideal world this would lead to you being able to compile the documents faithfully and report any shortcomings back to `tinytex`.  One can always hope...
